hydroflow_datalog_core/src/lib.rs

```diff
@@ -18,98 +18,142 @@ use util::{repeat_tuple, Counter};
 
 pub fn gen_hydroflow_graph(
     literal: proc_macro2::Literal,
-) -> Result<FlatGraph, (Vec<rust_sitter::errors::ParseError>, Vec<Diagnostic>)> {
+) -> Result<FlatGraph, Vec<Diagnostic>> {
     let str_node: syn::LitStr = parse_quote!(#literal);
     let actual_str = str_node.value();
-    let program: Program = grammar::datalog::parse(&actual_str).map_err(|e| (e, vec![]))?;
-
-    let mut inputs = Vec::new();
-    let mut outputs = Vec::new();
-    let mut rules = Vec::new();
-
-    for stmt in &program.rules {
-        match stmt {
-            Declaration::Input(_, ident) => inputs.push(ident),
-            Declaration::Output(_, ident) => outputs.push(ident),
-            Declaration::Rule(rule) => rules.push(rule),
+    let program: Program = grammar::datalog::parse(&actual_str);
+    
+    match program {
```

Comment:
We can probably simplify this by using `map_err`. So instead of having this match and nested code, instead

```rust
let program: Program = grammar::datalog::parse(&actual_str)
    .map_err(|errors| handle_errors(errors, &literal))?
```